### PR TITLE
CLISTUDIO-10128: If no server is created in start server/player, player sees spam messages about WaitForChild

### PR DESCRIPTION
**Pull request by Quenty**
It appears this also contains changes from a separate pull request. Looking into it.

### CODE REVIEW

**Comment by kyle-emmerich** [comment]
You'll probably just want to revert back to the commit before the one you want, stash the changes, and make a new branch, then apply the stashed changes on that branch.

File: CoreScriptsRoot/Modules/PlayerPermissionsModule.lua [hunk 1]
```diff
@@ -0,0 +1,41 @@
+local PlayerPermissionsModule = {}
+
+local function HasRankInGroupFunctionFactory(groupId, requiredRank)
+	local hasRankCache = {}
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/PlayerPermissionsModule.lua]: why don't we rely on the GetRankInGroup call to cache for us?

File: CoreScriptsRoot/Modules/PlayerlistModule.lua [hunk 2]
```diff
@@ -263,20 +266,15 @@ local function getFollowerStatusIcon(followerStatus)
...
-	end)
-	if not success then
-		print("PlayerListScript2: getAdminIcon() failed because", result)
-		return nil
-	end
+	if CUSTOM_ICONS[userIdStr] then return nil end
 	--
-	if result then
+	
+	if PlayerPermissionsModule.IsPlayerAdminAsync(player) then
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/PlayerlistModule.lua]: seems a bit weird to not use an actual function call here

File: CoreScriptsRoot/Modules/Chat.lua [hunk 3]
```diff
@@ -2215,7 +2233,7 @@ local function CreateChat()
 
 	function this:OnPlayerAdded(newPlayer)
 		if newPlayer then
-			spawn(function() Util.IsPlayerAdminAsync(newPlayer) end)
+			assert(coroutine.resume(coroutine.create(function() PlayerPermissionsModule.IsPlayerAdminAsync(newPlayer) end)))
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Chat.lua]: why did we switch from spawn?

File: CoreScriptsRoot/Modules/Chat.lua [hunk 4]
```diff
@@ -274,22 +275,34 @@ do
...
-		local userId = player and player.userId
-		if userId then
-			if adminCache[userId] == nil then
-				local isAdmin = false
-				-- Many things can error is the IsInGroup check
-				pcall(function()
-					isAdmin = player:IsInGroup(1200769)
-				end)
-				adminCache[userId] = isAdmin
+	local function MakeIsInGroup(groupId, requiredRank)
```
Comment (hockeychaos) [inline CoreScriptsRoot/Modules/Chat.lua]: this is really similar to the player permissions module function, can we use that instead?

**Comment by JetCrusherTorpedo** [comment]
Why not use `math.huge` for the timeout instead of `86400`?

File: CoreScriptsRoot/Modules/Chat.lua (hunk 4 above)
Comment (Quenty) [inline CoreScriptsRoot/Modules/Chat.lua]: This was from a previous merge, it's literally an older version of the code, should be cleaned up once I do what Kyle said to do.

**Comment by Quenty** [comment]
Waiting math.huge on ROBLOX returns instantly. 
`> wait(math.huge) print("Hi")`

**Comment by Quenty** [comment]
Ok. Closing this pull request in favor of another one without the unnecessary changes. (Thanks Kyle!)

**Pull request closed without merging**